### ISSUE TITLE
az eventgrid event-subscription list failing

### ISSUE THREAD

**Issue by ericthomas1** [Azure/azure-cli#26666]
### Describe the bug

Looks like someone else reported this same issue back in Dec. 2022: https://github.com/Azure/azure-cli/issues/24819

I'm facing the same issue today.

Error: `AttributeError: 'NoneType' object has no attribute 'lower'`

### Related command

`az eventgrid event-subscription list --source-resource-id "/subscriptions/<my-azure-subscription-id>/resourceGroups/<my-resource-group-name>/providers/Microsoft.Storage/storageAccounts/<my-storage-account-name>" --query "[].name" -o tsv`

### Errors

```
The command failed with an unexpected error. Here is the traceback:
'NoneType' object has no attribute 'lower'
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 1745, in cli_event_subscription_list
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 2616, in _list_event_subscriptions_by_resource_id
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 2669, in _validate_subscription_id_matches_default_subscription_id
AttributeError: 'NoneType' object has no attribute 'lower'
To check existing issues, please visit: https://github.com/Azure/azure-cli/issues
```

### Issue script & Debug output

Sanitized output:
```
cli.knack.cli: Command arguments: ['eventgrid', 'event-subscription', 'list', '--source-resource-id', 'C:/Program Files/Git/subscriptions/<my-azure-subscription-id>/resourceGroups/<my-resource-group-name>/providers/Microsoft.Storage/storageAccounts/<my-storage-account-name>', '--query', '[].name', '-o', 'tsv', '--debug']
cli.knack.cli: __init__ debug log:
Enable color in terminal.
cli.knack.cli: Event: Cli.PreExecute []
cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x0183B4F0>, <function OutputProducer.on_global_arguments at 0x01908C88>, <function CLIQuery.on_global_arguments at 0x0192A8E0>] 
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
cli.azure.cli.core: Modules found from index for 'eventgrid': ['azure.cli.command_modules.eventgrid']
cli.azure.cli.core: Loading command modules:
cli.azure.cli.core: Name                  Load Time    Groups  Commands       
cli.azure.cli.core: eventgrid                 0.009        25        96
cli.azure.cli.core: Total (1)                 0.009        25        96       
cli.azure.cli.core: Loaded 25 groups, 96 commands.
cli.azure.cli.core: Found a match in the command table.
cli.azure.cli.core: Raw command  : eventgrid event-subscription list
cli.azure.cli.core: Command table: eventgrid event-subscription list
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x03BCB418>]
cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to 'C:\Users\<my-windows-username>\.azure\commands\2023-06-13.11-25-59.eventgrid_event-subscription_list.24472.log'.
az_command_data_logger: command args: eventgrid event-subscription list --source-resource-id {} --query {} -o {} --debug
cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x03BF24F0>] 
cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x03C01418>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x03C01610>]
cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
cli.knack.cli: Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x01908CD0>, <function CLIQuery.handle_query_parameter at 0x0192A928>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x03C015C8>]
cli.azure.cli.core.commands.client_factory: Getting management service client client_type=EventGridManagementClient
cli.azure.cli.core.auth.persistence: build_persistence: location='C:\\Users\\<my-windows-username>\\.azure\\msal_token_cache.bin', encrypt=True
cli.azure.cli.core.auth.binary_cache: load: C:\Users\<my-windows-username>\.azure\msal_http_cache.bin
urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
msal.authority: openid_config = {'token_endpoint': 'https://login.microsoftonline.com/<my-aad-tenant-id>/oauth2/v2.0/token', 'token_endpoint_auth_methods_supported': ['client_secret_post', 'private_key_jwt', 'client_secret_basic'], 'jwks_uri': 'https://login.microsoftonline.com/<my-aad-tenant-id>/discovery/v2.0/keys', 'response_modes_supported': ['query', 'fragment', 'form_post'], 'subject_types_supported': ['pairwise'], 'id_token_signing_alg_values_supported': ['RS256'], 'response_types_supported': ['code', 'id_token', 'code id_token', 'id_token token'], 'scopes_supported': ['openid', 'profile', 'email', 'offline_access'], 'issuer': 'https://login.microsoftonline.com/<my-aad-tenant-id>/v2.0', 'request_uri_parameter_supported': False, 'userinfo_endpoint': 'https://graph.microsoft.com/oidc/userinfo', 'authorization_endpoint': 'https://login.microsoftonline.com/<my-aad-tenant-id>/oauth2/v2.0/authorize', 'device_authorization_endpoint': 'https://login.microsoftonline.com/<my-aad-tenant-id>/oauth2/v2.0/devicecode', 'http_logout_supported': True, 'frontchannel_logout_supported': True, 'end_session_endpoint': 'https://login.microsoftonline.com/<my-aad-tenant-id>/oauth2/v2.0/logout', 'claims_supported': ['sub', 'iss', 'cloud_instance_name', 'cloud_instance_host_name', 'cloud_graph_host_name', 'msgraph_host', 'aud', 'exp', 'iat', 'auth_time', 'acr', 'nonce', 'preferred_username', 'name', 'tid', 'ver', 'at_hash', 'c_hash', 'email'], 'kerberos_endpoint': 'https://login.microsoftonline.com/<my-aad-tenant-id>/kerberos', 'tenant_region_scope': 'NA', 'cloud_instance_name': 'microsoftonline.com', 'cloud_graph_host_name': 'graph.windows.net', 'msgraph_host': 'graph.microsoft.com', 'rbac_url': 'https://pas.windows.net'}
msal.application: Broker enabled? False
cli.azure.cli.core.azclierror: Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 1745, in cli_event_subscription_list
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 2616, in _list_event_subscriptions_by_resource_id
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 2669, in _validate_subscription_id_matches_default_subscription_id
AttributeError: 'NoneType' object has no attribute 'lower'

cli.azure.cli.core.azclierror: The command failed with an unexpected error. Here is the traceback:
az_command_data_logger: The command failed with an unexpected error. Here is the traceback:
cli.azure.cli.core.azclierror: 'NoneType' object has no attribute 'lower'     
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 1745, in cli_event_subscriptio  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 1745, in cli_event_subscription_list
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 2616, in _list_event_subscriptions_by_resource_id
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/eventgrid/custom.py", line 2669, in _validate_subscription_id_matches_default_subscription_id
AttributeError: 'NoneType' object has no attribute 'lower'
To check existing issues, please visit: https://github.com/Azure/azure-cli/issues
cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x03BCB538>]
az_command_data_logger: exit code: 1
cli.__main__: Command ran in 0.610 seconds (init: 0.294, invoke: 0.316)
telemetry.main: Begin splitting cli events and extra events, total events: 1    
telemetry.client: Accumulated 0 events. Flush the clients.
telemetry.main: Finish splitting cli events and extra events, cli events: 1     
telemetry.save: Save telemetry record of length 6969 in cache
telemetry.check: Negative: The C:\Users\<my-windows-username>\.azure\telemetry.txt was modified at 2023-06-13 11:17:47.331435, which in less than 600.000000 s
```

### Expected behavior

The list of Event Grid subscriptions for the given Storage Account would print to screen.

### Environment Summary

```
$ az version
{
  "azure-cli": "2.49.0",
  "azure-cli-core": "2.49.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {}
}
```

### Additional context

Please advise. Thank you.

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by navba-MSFT** [comment]
@ericthomas1 Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@ericthomas1 I was able to run the below command to list the subscription for Storage Account and it worked fine. Could you please  run the below command and check ?

> az login
az account set --subscription "1234-123-****-1234"
az eventgrid event-subscription list --source-resource-id /subscriptions/{SubID}/resourceGroups/{RG}/providers/Microsoft.Storage/storageaccounts/MyStorageName

I used the same AzCLI version as yours (2.49.0. Please let me know if you face any issues. Awaiting your reply.

**Comment by ericthomas1** [comment]
Thank you @navba-MSFT. You helped me figure out what my issue was. Your suggestion of first `az login` then `az account set` showed me I was using the wrong subscription id in my original command.

`az login` yielded:

```
[
  {
    "cloudName": "<azure-cloud-name>",
    "homeTenantId": "<home-tenant-id>",
    "id": "<id (of what?! aha...its the subscription-id)>",
    "isDefault": true,
    "managedByTenants": [
      {
        "tenantId": "<managed-by-tenant"
      }
    ],
    "name": "<az-subscription-name>",
    "state": "Enabled",
    "tenantId": "<same-as-home-tenant-id>",
    "user": {
      "name": "<my-name>@<my-org.tld>",
      "type": "<user-type>"
    }
  },
  {
    "cloudName": "<azure-cloud-name>",
    "homeTenantId": "<home-tenant-id>",
    "id": "<2nd-az-subscription-tenant-id>",
    "isDefault": false,
    "managedByTenants": [],
    "name": "<2nd-az-subscription-name>",
    "state": "Enabled",
    "tenantId": "<home-tenant-id>",
    "user": {
      "name": "<my-name>@<my-org.tld>",
      "type": "<user-type>"
    }
  }
]
```

In my original command, the one that caused me to open this ticket, I used:

`az eventgrid event-subscription list --source-resource-id "/subscriptions/<home-tenant-id>/resourceGroups/<my-resource-group-name>/providers/Microsoft.Storage/storageAccounts/<my-storage-account-name>" --query "[].name" -o tsv`

By working through your suggestion, specifically `az account set`, I realized that `<home-tenant-id>` was not the correct value. I actually needed to use the `<id>` value in the original command.

Basically "`id`" is the "`subscription`" value I should have used :).

**Comment by ericthomas1** [comment]
I was just receiving this error again, this time I had the above dialed in. I realized that this was an error in the `bash` script. When I converted the script to PowerShell, I no longer received the `AttributeError: 'NoneType' object has no attribute 'lower'` error!
